scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java

```diff
@@ -236,20 +238,20 @@ private List<AssociatedArtifact> extractAssociatedArtifactsFromDoi(DocTp docTp)
         }
     }
 
-    private ScopusFile fetchFileContent(ScopusFile file) {
+    private ScopusFile fetchFileContent(ScopusFile file) throws IOException {
         var fetchFileResponse = fetchResponseAsInputStream(file.downloadFileUrl());
-        var content = fetchFileResponse.body();
         var filename = getFilename(fetchFileResponse);
-
+        var size = fetchFileResponse.body().readAllBytes().length;
         return file.copy()
-                   .withMimeType(extractMimeType(file, fetchFileResponse))
-                   .withContent(content).withName(filename).build();
+                   .withMimeType(getMimeType(fetchFileResponse.body()))
+                   .withContent(fetchFileResponse.body())
+                   .withSize(size)
+                   .withName(filename).build();
     }
 
-    private static String extractMimeType(ScopusFile file, HttpResponse<InputStream> fetchFileResponse) {
-        return hasSupportedMimeType(file)
-                   ? file.mimeType()
-                   : fetchFileResponse.headers().firstValue(Header.CONTENT_TYPE).orElse(null);
+    private String getMimeType(InputStream inputStream) throws IOException {
+        return TikaConfig.getDefaultConfig().getDetector()
+                   .detect(new BufferedInputStream(inputStream), new Metadata()).toString();
```

Comment:
Problem:

Headers often miss content-type header, creating s3 file without mimeType result in default mimeType application/octet-stream. Our endpoint downloads this file instead of showing it. Tika seems to do what we need with minimum code.